File: Find the Student that Will Replace the Chalk.js
var chalkReplacer = function(chalk, k) {
var  i = 0;
    while(i<=chalk.length){
        if(i === chalk.length){
            i = 0;
        } else if(k >= chalk[i]){
            k -= chalk[i];
            i++;
        } else {
            return i;
        }
    }
            
    
   

};

File: 2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        total_sum = sum(chalk)
        k%=total_sum
        for i in range(len(chalk)):
            if chalk[i] > k:
                return i
            k-=chalk[i]
        return -1
       